### PR TITLE
Removal of Japanese text

### PR DESCRIPTION
**Pull request by LanHikariDS**
Somewhat self explanatory, makes the Status screen and it's relevant information readable in English.
Shinbok-0

~However, there are three issues with my implementation. For one, the Sunrise/Sunset text doesn't give enough room to put the text there, and the engine does some icon switching based on which line to use for the current time. As a result, I made two icons to represent that line,~

~Secondly, I had to modify the inventory screens slightly to free up tiles for the extra text. As a result, the dark brown bar to the right of the player's inventory now reaches the edge of the screen, whereas it didn't before.~
Shin Bokura no Taiyou - Gyakushuu no Sabata (Japan)-0 Shinbok-1

~Finally, and most importantly, when in a shop, the top bar reads as "TIMN", even though I've changed the map to say "TIME". For reasons I don't quite understand yet, the engine is swapping out these tiles on the map, and I don't know how to disable it. The best I could do is figure out what tile is getting copied to both the N in BANK and the E in TIME, but changing that tile to an E will change BANK with it.~
Shinbok-2

This PR is still under development, and is not ready to be merged.

### CODE REVIEW

**Comment by LanHikariDS** [comment]
Translated a lot of the button prompts, ~though they need deeper work. A few of them cut off or pull tiles from other buttons, and the END button on text entry doesn't animate in the bottom corners, oddly enough.~
Shinbok-5
Shinbok-6
Shinbok-7
Shinbok-3

**Comment by LanHikariDS** [comment]
Fixed the issue with the END label highlight, as well as Status Point highlights (which I hadn't mentioned prior).
-Translated the password and name entry headers. ~Bike Name's positioning needs updated.~
-Updated the END button to be closer to Boktai 2's.
-Gave the CUSTOM menu a basic translation, ~though as a result, the Course Select menu is filled with errors, hope to correct later.~
-Corrected Cheyenne's shop label to no longer say "Sheyenne"
Shinbok-9
Shinbok-10
Shinbok-8
Shinbok-11

**Comment by LanHikariDS** [comment]
~Whoops~
image

**Comment by LanHikariDS** [comment]
I learned what menus.ips was, since files.txt doesn't make any mention of it, and it seems to be Spikeman's attempt at what I did in my first commit, though he seems to have done a much better job than I did, using full words for the sunrise/sunset clock, and not having the issue with the N duplicating. As such, I rebuilt my changes on top of his patch, and redid a few of my own stylistic choices, such as the longer STATION text, and using SPT for Spirit, instead of SPR, to distance it from Strength. Not sure why it wasn't implemented into main_translation.ips, though...

**Comment by LanHikariDS** [comment]
Also went ahead and fixed the ticks on the Config menu
ShinbokTrans-0

**Comment by LanHikariDS** [comment]
The Circuit menu is finally translated, which entailed properly redoing the Custom menu and polishing it up from it's previous incarnation. Barring any major issues, the next push _should_ just need some bug testing, and then this PR will be ready to merge.

image

**Comment by LanHikariDS** [comment]
Cool, I KNOW I fixed the Bike Results screen, but now it's just magically broken again

**Comment by LanHikariDS** [comment]
Translated all of the Library Guides, which now means all graphical text except for the header on Dark Loans and Solar Bank is translated. This PR is definitely coming to a close, but still needs some tidying up, namely the number after "Page" in the Album and Dolls menu needs repositioned, but I'm currently struggling with that, as it's positioned by a huge pain-in-the-ass function. Also fixed a bug in the Monster>Item display where "Normal:" would word wrap early, as well as covering up the first letter of the Normal item drop's name.

ShinbokTrans-18
ShinbokTrans-19
ShinbokTrans-20
ShinbokTrans-21

**Comment by LanHikariDS** [comment]
Gonna mark it as ready for review just to open it up. The page number is still being a massive pain in the ass, and is an incredibly minor issue, so I'll fix it in a later PR if this gets merged. Current build merges Status_Screen.ips' changes into the 'master' patches.

**Comment by LanHikariDS** [comment]
TLDR of the PR- Feature creep is a bitch, and _most_ of the game's text is now in English, as well as offering a second script option that keeps various names closer to the official English translation, on top of a few typo/bug fixes.

Would recommend heavy testing before merging, as I haven't checked too deeply for anything that was accidentally/consequentally changed in the process.

**Comment by LanHikariDS** [comment]
Nope, back to the drawing board. Realized the final end screen was _never_ looked at until I was testing something else, so that still needs corrected. At least it seems to be somewhere with free ROM space
image

**Comment by Neo-Qwerty** [comment]
I know that this is very far from timely but IIRC the menus.ips patch broke new game+ (Edit: at least on the versions of VBA that were around circa 2011), and that's why it wasn't merged into the main translation patch (it wasn't just a "mere" case of the end results screen being graphically broken). 

This is based on my admittedly vague memories of the warnings being passed around on the old San Miguel forum and my ROM folder from 2009 having both patched versions, just wanted to make this a note for testing.

**Comment by LanHikariDS** [comment]
I appreciate it, this is the first I've heard of the issue, and I'll keep an eye out for it
> it wasn't just a "mere" case of the end results screen being graphically broken

As for the Result screen, that has nothing to do with menus.ips, that's something _I_ fucked up in the process of translating other stuff, as this PR goes above and beyond what Menus.ips did

**Comment by TheArkhomDestroyer** [comment]
Any word on progress?

**Comment by LanHikariDS** [comment]
Progress is moving along somewhat slowly behind the scenes, as I've been focusing on other areas of the game, as well as personal issues coming up this past week keeping me occupied. I've fixed the above Result text internally, but I've been holding off on pushing it, as a few bugs have also come to my attention I would like to iron out first. Also I've been working on finishing up the credits, so the ideal is that the next commit will be the removal of all Japanese text in the game.

**Comment by LanHikariDS** [comment]
Opening this back up for testing and review, as this update marks the removal of _basically_ all Japanese text from the game (minus the Password screen), and will likely be the last update I post until 1.0 is ready.

**Comment by F4BB** [comment]
> Opening this back up for testing and review, as this update marks the removal of _basically_ all Japanese text from the game (minus the Password screen), and will likely be the last update I post until 1.0 is ready.

A small question, after the exchange on Twitter (it’s me who translate the game in French).
I have a question about the file, did you change the encoding format of scripts.sjs ? 

In French there are so many (too much clearly) accent but format file does not take accents into account during injection.

I try to change format but it’s break everything. :,(

**Comment by LanHikariDS** [comment]
> > Opening this back up for testing and review, as this update marks the removal of _basically_ all Japanese text from the game (minus the Password screen), and will likely be the last update I post until 1.0 is ready.
> 
> A small question, after the exchange on Twitter (it’s me who translate the game in French). I have a question about the file, did you change the encoding format of scripts.sjs ?
> 
> In French there are so many (too much clearly) accent but format file does not take accents into account during injection.
> 
> I try to change format but it’s break everything. :,(

Did the accents work before my newest update? I know Boktai 3 doesn't have a full set of accentives present when I checked, but I didn't change anything like that personally. Maybe Github screwed something up? I did have to use a different upload process this time around that maybe broke something there...

**Comment by F4BB** [comment]
> > > Opening this back up for testing and review, as this update marks the removal of _basically_ all Japanese text from the game (minus the Password screen), and will likely be the last update I post until 1.0 is ready.
> > 
> > 
> > A small question, after the exchange on Twitter (it’s me who translate the game in French). I have a question about the file, did you change the encoding format of scripts.sjs ?
> > In French there are so many (too much clearly) accent but format file does not take accents into account during injection.
> > I try to change format but it’s break everything. :,(
> 
> Did the accents work before my newest update? I know Boktai 3 doesn't have a full set of accentives present when I checked, but I didn't change anything like that personally. Maybe Github screwed something up? I did have to use a different upload process this time around that maybe broke something there...

After many try, with old file and new file released, it's seem the Shift-JIS not support the french accent like "é,à,è,ç".
I try many other encoding but all failed. I continue to search a solution on this way.For everything else all your work is amazing and help me so much ! Thanks Again.

**Comment by LanHikariDS** [comment]
> After many try, with old file and new file released, it's seem the Shift-JIS not support the french accent like "é,à,è,ç". I try many other encoding but all failed. I continue to search a solution on this way.For everything else all your work is amazing and help me so much ! Thanks Again.

So even in the prior update, these accents didn't work? I can see that the "é,à,è,ç" characters _are_ in the game's font table, I'll give them a try locally and see if I can get them to display.

**Comment by raphaelr** [comment]
For Boktai 1, the diacritics are encoded like this (didn't try in Boktai 3):
<details>

```
1F4C=À
1F68=Á
1F66=Â
1F01=Ä
1F03=Ç
1F6A=È
1F04=É
1F67=Ê
1F69=Ë
1F6E=Ì
1F6B=Í
1F6C=Î
1F6D=Ï
1F72=Ò
1F6F=Ó
1F70=Ô
1F06=Ö
1F75=Ù
1F73=Ú
1F74=Û
1F07=Ü
1F5A=Ý
1F28=ß
1F09=à
1F08=á
1F0A=â
1F0D=å
1F0B=ä
1F0E=ç
1F10=è
1F0F=é
1F11=ê
1F12=ë
1F14=ì
1F13=í
1F15=î
1F16=ï
1F19=ò
1F18=ó
1F1A=ô
1F1B=ö
1F1E=ù
1F1D=ú
1F1F=û
1F20=ü
1F58=ý
1F59=ÿ
1F48=«
1F49=»
1F22=°
1F42=¡
1F41=¿
1F17=ñ
1F05=Ñ
```

</details>

**Comment by LanHikariDS** [comment]
> For Boktai 1, the diacritics are encoded like this (didn't try in Boktai 3):

The few characters I tested in Shinbok still seem to line up. @F4BB So unfortunately, it seems like the script editor ignores changes with diacretics in the script proper, but you can still force it using the hex codes @raphaelr posted above in the script. In the below screenshot, that line in the script currently is written in the SJS as:
> M{1F}{09}k{1F}{0F} sunlight hit th{1F}{10}
> Sol{1F}{08}r Sensor on the
> Game Pak & you {1F}{0E}an enjoy
> many effects in BOKTAI3.
ShinbokAccent-1

EDIT: I dunno why/how the sprites are screwed up in that screenshot, but that doesn't seem to actually be the case in-game...

**Comment by F4BB** [comment]
Thanks @LanHikariDS @raphaelr ! Your help me so much ! with HEX codes it works perfectly !

Now i work on it !

2025-02-07 08_48_00-mGBA - BOKTAI3 (240 7 fps) - 0 10 2
2025-02-07 08_49_52-mGBA - BOKTAI3 - 0 10 2


> > For Boktai 1, the diacritics are encoded like this (didn't try in Boktai 3):
> 
> The few characters I tested in Shinbok still seem to line up. @F4BB So unfortunately, it seems like the script editor ignores changes with diacretics in the script proper, but you can still force it using the hex codes @raphaelr posted above in the script. In the below screenshot, that line in the script currently is written in the SJS as:
> 
> > M{1F}{09}k{1F}{0F} sunlight hit th{1F}{10}
> > Sol{1F}{08}r Sensor on the
> > Game Pak & you {1F}{0E}an enjoy
> > many effects in BOKTAI3.
> > ShinbokAccent-1
> 
> EDIT: I dunno why/how the sprites are screwed up in that screenshot, but that doesn't seem to actually be the case in-game...

**Comment by LanHikariDS** [comment]
@F4BB I also ask that you post any other issues with your translation in an issue thread in the original translation's Github, as it's not directly related to this PR

**Comment by LanHikariDS** [comment]
Bahhhh, the Sleep menu needs touched up a bit, completely forgot about that one

ShinbokTrans-48

**Comment by F4BB** [comment]
> Bahhhh, the Sleep menu needs touched up a bit, completely forgot about that one
> 
> ShinbokTrans-48

Personaly in this case and in French i just "wrote" this equivalent  ----   "Press SELECT + L + R to wake UP"

I win space and character in the scripts.

**Comment by LanHikariDS** [comment]
> Personaly in this case and in French i just "wrote" this equivalent ---- "Press SELECT + L + R to wake UP"
> 
> I win space and character in the scripts.

I need to check how that screen looks in Boktai 2, but most likely I'm just gonna redefine the bounds on that text box and reposition it accordingly

EDIT: I've fixed it locally, but I don't feel like fussing with making the .ips files tonight. This can be corrected by changing the first instance of `04 21 06 22 16` in the ROM (Located at 0x08176070) to `02 21 06 22 1A`

**Comment by lecarde17** [comment]
Hi there. Really wanna help you with finishing translating the rest of the text, I'm decent with graphics editing software.
Please contact me on Discord (marklecarde) if interested, I'm really eager to help.
Always wanted to work on this game!

**Comment by LanHikariDS** [comment]
> Hi there. Really wanna help you with finishing translating the rest of the text, I'm decent with graphics editing software. Please contact me on Discord (marklecarde) if interested, I'm really eager to help. Always wanted to work on this game!

I appreciate your offer, but we don't have need for that. We're pretty much done with all the graphics stuff, we're just doing some final polishing on the code and some other things.

**Comment by lecarde17** [comment]
> > Hi there. Really wanna help you with finishing translating the rest of the text, I'm decent with graphics editing software. Please contact me on Discord (marklecarde) if interested, I'm really eager to help. Always wanted to work on this game!
> 
> I appreciate your offer, but we don't have need for that. We're pretty much done with all the graphics stuff, we're just doing some final polishing on the code and some other things.

Aight, I understand, thanks for telling me. Good luck to your project.

**Comment by CarterLi** [comment]
Hello there. After applying the patch `latest_translation.ips`, neither mgba nor vbam can play the game. They just freeze at boot.

The old VBA works though

image

**Comment by F4BB** [comment]
Hey,
Thanks again, LanHikariDS, raphaelr, for your amazing work!

To make sure I understand how the region-menu-eu works, I tried several things.

First, I used the IPS patch on my version, but it didn’t work. Then I tried using the two scripts: region.sjs and scripts.sjs.

I replaced the relevant lines in scripts.sjs with the ones I found in region.sjs, but I didn’t change the line $X=====9340=====.
If I change it to $E=====9340=====, the game crashes. But if I leave it as $X=====9340=====, the EU region menu appears — however, the red selection border doesn’t move up or down.

For now, I can play and continue my translation, but the region selection menu doesn’t work 100%. It’s not a blocker, and if I manage to get it working, it’ll just be a nice little bonus.
There’s no urgency — I’m just sharing my feedback here.

**Comment by julianmpineda** [comment]
A bit random, but early on when Otenko describes levelling up and skill points he calls the last stat "skill" but it should be "strength" to match the "STR" on the pause stat screen.

**Comment by LanHikariDS** [comment]
> A bit random, but early on when Otenko describes levelling up and skill points he calls the last stat "skill" but it should be "strength" to match the "STR" on the pause stat screen.

Ah, in 8777, I see it. I'll go ahead and correct it locally, but it might not show up in the repo for a while, since I'm trying to hold off on updates until 1.0

**Comment by Asilver-jpg** [comment]
Hey there! Huge fan of the first two games. Do you guys need help with anything?
